pact/coin-contract/coin.repl

```diff
@@ -452,3 +452,75 @@
 (expect-failure
   "bad redeem-gas fails"
   (redeem-gas 'doug (read-keyset 'doug) 'emily UNIT_BAD))
+
+(commit-tx)
+
+;; Coin allocation tests
+
+(begin-tx)
+
+(use coin)
+
+; account creation
+
+(expect-failure
+  "allocation account creation only occurs at genesis"
+  (create-allocation-account "brandon" (time "1900-10-31T00:00:00Z") "brandon" 200000.0))
+
+(test-capability (GENESIS))
+
+(expect-failure
+  "all allocation amounts must be positive"
+  (create-allocation-account "brandon" (time "1900-10-31T00:00:00Z") "brandon" -200000.0))
+
+(expect-failure
+  "all allocation accounts must satisfy coin contract account min chars"
+  (create-allocation-account "br" (time "1900-10-31T00:00:00Z") "brandon" 200000.0))
+
+(expect-failure
+  "all allocation accounts must satisfy coin contract account max chars"
+  (create-allocation-account "There he met Saunders Mac Lane. Mac Lane, then visiting Paris, was anxious to learn from Yoneda, and commenced an interview with Yoneda in a café at Gare du Nord. The interview was continued on Yoneda’s train until its departure. In its course, Mac Lane learned about the lemma and subsequently baptized it." (time "1900-10-31T00:00:00Z") "brandon" 200000.0))
+
+
+; (expect-failure
+;  "keyset-ref guards ..."
+;  (create-allocation-account "brandon" (time "2020-10-31T00:00:00Z") "brandon" 200000.0))
+
+; successful keyset refs require defined keyset
+
+(env-data { "brandon" : ["brandon"]})
+(define-keyset "brandon" (read-keyset "brandon"))
+
+(expect
+  "allocating coin accounts succeeds"
+  "Write succeeded"
+  (create-allocation-account "brandon" (time "2020-10-31T00:00:00Z") "brandon" 10.0))
+
+; release-allocation
+
+(expect-failure
+  "allocation release fails when keys are not in scope"
+  (release-allocation "brandon"))
+
+(env-keys ["brandon"])
+
+(expect-failure
+  "allocation coins fails since release date is not >= current time"
+  (release-allocation "brandon"))
+
+(env-chain-data { "block-time" : (time "2020-10-31T00:00:00Z") })
+
+(expect
+  "successfully allocates funds for correct amounts and date"
+  "Allocation successful"
+  (release-allocation "brandon"))
+
+(expect-failure
+  "releases fail when funds have been redeemed"
+  (release-allocation "brandon"))
+
+(expect
+  "brandon has 10 coins released to his account in coin contract"
+  "brandon has 10.0 coins guarded by 'brandon"
```

Comment:
I would just do equality on the info object. Keyset ref guard equality works

Reply:
Done